app.py

```diff
@@ -434,7 +434,18 @@ def prepare_generic_launch(self, engine_name, context, app_path, app_args):
         sys.path.insert(0, python_path)
         try:
             import bootstrap
-            (app_path, new_args) = bootstrap.bootstrap(engine_name, context, app_path, app_args, self)
+            extra_args = self.get_setting("extra", {})
+
+            # bootstrap should take kwargs in order to protect from changes in
+            # this signature in the future.  For example:
+            # def bootstrap(engine, context, app_path, app_args, **kwargs)
+            (app_path, new_args) = bootstrap.bootstrap(
+                engine_name=engine_name,
+                context=context,
+                app_path=app_path,
+                app_args=app_args,
+                extra_args=extra_args,
```

Comment:
The trailing comma is intentional, just to make copy/pasting/reordering the args easier (reordering is more for dictionaries, but still).  Alan was fine leaving it as is.